### DISCUSSION TITLE
Is the TreeControl widget incompatible with the css branch?

### DISCUSSION THREAD

**Discussion by jeffwright13** [Textualize/textual#364] [Q&A]
I'm trying to re-use some code I had written for the main branch, which uses TreeControl widgets. Whenever I try to 'instantiate' a TreeControl object, I get a traceback that says:
```
  File "/Users/jwr003/coding/pytest-fold/pytest_fold/tui_textual_tab.py", line 74, in on_mount
    self.fail_tree = TreeControl(
  File "/Users/jwr003/coding/pytest-fold/venv/lib/python3.10/site-packages/textual/widgets/_tree_control.py", line 184, in __init__
    self.id = NodeID(0)
  File "/Users/jwr003/coding/pytest-fold/venv/lib/python3.10/site-packages/textual/dom.py", line 114, in id
    if self._id is not None:
AttributeError: 'TreeControl' object has no attribute '_id'. Did you mean: 'id'?
```
I did a bit of digging in the code, and it looks like the underlying Widget class has been changed. It no longer inherits from `MessagePump`; instead, it now inherits from a new object called `DOM`, which apparently has a `_id` attribute instead of `id`. The `_tree_control.py` file remains the same, though, so its API is incompatible.

I know the code is still in "alpha" but I sure would like to use some of the new widgets available in the css branch (esp. the Tabs widget). Is there any way to hack this up without too much trouble?

**Comment by willmcgugan** [comment]
Sorry Jeff. We’re super actively working on this. Can’t guarantee things will work in the short term. Code is in the oven, but it’s not quite fully baked.

**Reply by jeffwright13** [reply]
OK, thanks. I'll see what I can work out. Some ideas are percolating thru my head.

**Reply by jeffwright13** [reply]
Or if you and your team want to make a quick handful of $$$ maybe I could contract you for what is likely a simple app. :-)

**Reply by willmcgugan** [reply]
Drop me a line on will@textualize.io and we’ll arrange a zoom call.
